src/commands/test/lib/test-cloud-uploader.ts

```diff
@@ -123,52 +124,44 @@ export class TestCloudUploader {
     });
   }
 
-  private async uploadHashOrNewFile(testRunId: string, file: TestRunFile): Promise<void> {
-    if (await this.tryUploadFileHash(testRunId, file)) {
-      debug(`File ${file.sourcePath}: hash upload`);
-    }
-    else {
-      await this.uploadFile(testRunId, file);
-      debug(`File ${file.sourcePath}: direct upload`);
-    }
+  private async uploadFilesUsingBatch(testRunId: string, files: TestRunFile[]): Promise<void> {
+    let checkHashResult = await this.uploadHashesBatch(testRunId, files.map(f => { return { file: f }; }));
+
+    let limit = pLimit(paralleRequests);
+    let uploadNewFilesTask = checkHashResult
```

Comment:
If this is an array, it should probably have a plural name - `uploadNewFilesTasks` instead?